src/java/org/apache/cassandra/io/sstable/ReducingKeyIterator.java

```diff
@@ -33,7 +33,7 @@
 public class ReducingKeyIterator implements CloseableIterator<DecoratedKey>
 {
     private final ArrayList<KeyIterator> iters;
-    private IMergeIterator<DecoratedKey,DecoratedKey> mi;
+    private volatile IMergeIterator<DecoratedKey,DecoratedKey> mi;
```

Comment:
Nit: missed whitespace
```suggestion
    private volatile IMergeIterator<DecoratedKey, DecoratedKey> mi;
```
```suggestion
    private volatile IMergeIterator<DecoratedKey,DecoratedKey> mi;
```